akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -230,6 +250,29 @@ private[http2] class Http2ServerDemux(http2Settings: Http2CommonSettings, initia
         }
       })
 
+      /**
+       * Tune this peer to enforce the settings configured from this peer.
+       * @return TODO: it's a Boolean but I don't think it has to be.
+       */
+      private def enforceSettings(settings: immutable.Seq[Setting]): Boolean = {
+        var settingsAppliedOk = true
+
+        settings.foreach {
+          case Setting(Http2Protocol.SettingIdentifier.SETTINGS_MAX_CONCURRENT_STREAMS, value) =>
+          // nothing to do here: the count of open incoming streams
+          // depends on `def maxConcurrentStreams` above which is currently not modifiable
+        }
```

Comment:
The default `SETTINGS_MAX_CONCURRENT_STREAMS` is _infinite_ and we're enforcing it to `max-concurrent-streams` at construction time already. That could be improved if we enabled the _enforcing_ in this piece of code. Not sure the added complexity is worth it at this point.

Reply:
```suggestion
          case Setting(Http2Protocol.SettingIdentifier.SETTINGS_MAX_CONCURRENT_STREAMS, value) =>
          // TODO: Http2Compliance.compliesWithMaxConcurrentStreams should only be enabled once 
          //  we get this SETTINGS_ACK.
        }
```